matchms/Pipeline.py

```diff
@@ -0,0 +1,367 @@
+import logging
+import os
+from collections import OrderedDict
+import yaml
+from tqdm import tqdm
+import matchms.filtering as msfilters
+import matchms.importing as msimport
+import matchms.similarity as mssimilarity
+from matchms import calculate_scores
+from matchms.logging_functions import (add_logging_to_file,
+                                       set_matchms_logger_level)
+
+
+_importing_functions = {"json": msimport.load_from_json,
+                        "mgf": msimport.load_from_mgf,
+                        "msp": msimport.load_from_msp,
+                        "mzml": msimport.load_from_mzml,
+                        "mzxml": msimport.load_from_mzxml}
+_filter_functions = {key: f for key, f in msfilters.__dict__.items() if callable(f)}
+_masking_functions = ["filter_by_range"]
+_score_functions = {key.lower(): f for key, f in mssimilarity.__dict__.items() if callable(f)}
+logger = logging.getLogger("matchms")
+
+
+class Pipeline:
+    """Central pipeline class.
+
+    The matchms Pipeline class is meant to make running extensive analysis pipelines
+    fast and easy. I can be used in two different ways. First, a pipeline can be defined
+    using a config file (a yaml file, best to start from the template provided to define
+    your own pipline).
+
+    Once a config file is defined, execution only needs the following code:
+
+    .. code-block:: python
+
+        from matchms import Pipeline
+
+        pipeline = Pipeline("my_config_file.yaml")
+        pipeline.run()
+
+    The second way to define a pipeline is via a Python script. The following code is an
+    example of how this works:
+
+    .. code-block:: python
+
+        from matchms import Pipeline
+
+        pipeline = Pipeline("my_config_file.yaml")
+        pipeline.run()
+
```

Comment:
There seems to be a repeated example in docstrings.

Reply:
Thanks, should be fixed now.